guides/cookbook.md

```diff
@@ -579,6 +780,78 @@ examining the type annotations of each top-level struct and comparing against
 not match `"foo"`, the reader can quickly skip to the start of the next value.
 
 <script>writeTabs()</script>
+<div class="tabpane C" markdown="1">
+```c
+#include <cstdlib>
+#include "ionc/ion.h"
+
+#define ION_OK(x) if (x) { printf("Error: %s\n", ion_error_to_str(x)); return x; }
+
+void ion_string_from_cstr(const char *cstr, ION_STRING *out) {
+    out->value = (BYTE *)cstr;
+    out->length = (SIZE)strlen(cstr);
+}
+
+int main(int argc, char **argv) {
+    const char ion_binary[] = { ... };    // bytes representing the Ion binary encoding of the above data
```

Comment:
Are we ok with examples in the cookbook that cannot be executed with a copy-paste?